### DISCUSSION TITLE
Streaming content, append table row

### DISCUSSION THREAD

**Discussion by tuxcanfly** [alpinejs/alpine#716] [4. General]
I need to dynamically append a row to a table:

https://codepen.io/tuxcanfly/pen/XWdjEyd

I tried x-model="rows" but it didn't work. Any idea how I can do this?

<p class="codepen" data-height="265" data-theme-id="light" data-default-tab="html,result" data-user="tuxcanfly" data-slug-hash="XWdjEyd" style="height: 265px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="alpine.js streaming table">
  <span>See the Pen <a href="https://codepen.io/tuxcanfly/pen/XWdjEyd">
  alpine.js streaming table</a> by Javed (<a href="https://codepen.io/tuxcanfly">@tuxcanfly</a>)
  on <a href="https://codepen.io">CodePen</a>.</span>
</p>
<script async src="https://static.codepen.io/assets/embed/ei.js"></script>

**Comment by KevinBatdorf** [comment]
One thing that stands out is that pen is using Alpine 1.9.8 which is quite old already.

Update the script to:

https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.x.x/dist/alpine.min.js

Also, you need to call your setInterval function inside of x-init for the component to update

```html
<tbody x-data="handler()" x-init="setInterval(() => {
    rows.push({
        id: rows.length + 1,
        msg: 'hello world',
        timestamp: '20 Aug 2020'
    })
}, 1000)">
```

**Reply by ryangjchandler** [reply]
@tuxcanfly - it's worth noting that `x-model` is for two-way data binding on form inputs, not layout elements such as `<table>`.
